crates/cli/src/opts/build/core.rs

```diff
@@ -167,21 +167,24 @@ impl CoreBuildArgs {
 // Loads project's figment and merges the build cli arguments into it
 impl<'a> From<&'a CoreBuildArgs> for Figment {
     fn from(args: &'a CoreBuildArgs) -> Self {
-        let mut figment = if let Some(ref config_path) = args.project_paths.config_path {
+        let (mut figment, config) = if let Some(ref config_path) = args.project_paths.config_path {
             if !config_path.exists() {
                 panic!("error: config-path `{}` does not exist", config_path.display())
             }
             if !config_path.ends_with(Config::FILE_NAME) {
                 panic!("error: the config-path must be a path to a foundry.toml file")
             }
             let config_path = canonicalized(config_path);
-            Config::figment_with_root(config_path.parent().unwrap())
+            let root_path = config_path.parent().unwrap();
+            (Config::figment_with_root(root_path), Config::load_with_root(root_path))
```

Comment:
this is very expensive and now does `Config::figment_with_root(root_path)` twice under the hood.

especially because this is only used to apply smth to arguments?
feels like the wrong location to do this

Reply:
looks like we only need the project paths and not the entire config to modify the remappings?

Reply:
yep, good point, I changed this in https://github.com/foundry-rs/foundry/pull/9258/commits/8b2d7b026ac6a5e420fcb472139e394e9126754c to load them from figment, this also correctly loads the current profile src/script/test dirs.
Hope this makes sense, pls check. thanks!